plugins/kobolight.koplugin/main.lua

```diff
@@ -72,6 +72,12 @@ end
 function KoboLight:resetLayout()
     local new_screen_height = Screen:getHeight()
     self.gestureScale = new_screen_height * swipe_touch_zone_ratio.h * 0.8
+    local powerd = Device:getPowerDevice()
+    local scale = (powerd.fl_max - powerd.fl_min) / 20
+    for i = 1, #self.steps, 1
+    do
+        self.steps[i] = math.floor(self.steps[i] * scale)
```

Comment:
The /20 means /2 /10, i.e. swiping twice can reach the maximum or minimum value, since 10 is the maximum one in steps.
I change the logic to calculate the steps to ensure a smooth curve.